server/cmwell-data-tools/src/main/scala/cmwell/tools/data/sparql/SparqlTriggeredProcessor.scala

```diff
@@ -47,18 +47,37 @@ case class Config(name: Option[String] = None,
                   hostUpdatesSource: Option[String],
                   force: Option[Boolean] = Some(false))
 
-object SparqlTriggeredProcessor {
+object SparqlTriggeredProcessor extends DataToolsLogging {
 
   val sparqlMaterializerLabel = "sparql-materializer"
 
+  def loadInitialTokensAndStatistics(tokenReporter : Option[ActorRef])
+                                    (implicit system: ActorSystem, mat: Materializer, ec: ExecutionContext) = tokenReporter match {
+
+
+    case None => Right(AgentTokensAndStatistics(Map.empty[String, TokenAndStatistics], None, None))
+    case Some(reporter) =>
+      import akka.pattern._
+      implicit val t = akka.util.Timeout(1.minute)
+      val result = (reporter ? RequestPreviousTokens)
+        .mapTo[ResponseWithPreviousTokens]
+        .map {
+          case ResponseWithPreviousTokens(tokens) => tokens
+          case x => logger.error(s"did not receive previous tokens: $x"); Left(s"did not receive previous tokens: $x")
+        }
+      Await.result(result, 1.minute)
```

Comment:
The rationale behind this is that the above function returns a AgentTokensAndStatistics. This is used by the ingest as well as the listen flows, and I thought it would be sensible to load this once. I understand your point about the whole manager crashing in this scenario. I am looking into a more graceful fallback.

Obviously it should be a lot more resilient as we are not using infotons for token storage, but we still need to check.

Reply:
Believe I have resolved the issue surrounding the whole manager crashing if the call fails in my latest commit